packages/astro/src/content/vite-plugin-content-virtual-mod.ts

```diff
@@ -253,18 +259,24 @@ export async function generateContentEntryFile({
 		renderEntryGlobResult = getStringifiedCollectionFromLookup('render', relContentDir, lookupMap);
 	}
 
-	let virtualModContents =
-		nodeFs
-			.readFileSync(contentPaths.virtualModTemplate, 'utf-8')
-			.replace('@@CONTENT_DIR@@', relContentDir)
-			.replace("'@@CONTENT_ENTRY_GLOB_PATH@@'", contentEntryGlobResult)
-			.replace("'@@DATA_ENTRY_GLOB_PATH@@'", dataEntryGlobResult)
-			.replace("'@@RENDER_ENTRY_GLOB_PATH@@'", renderEntryGlobResult)
-			.replace('/* @@LOOKUP_MAP_ASSIGNMENT@@ */', `lookupMap = ${JSON.stringify(lookupMap)};`) +
-		(isClient
-			? `
-console.warn('astro:content is only supported running server-side. Using it in the browser will lead to bloated bundles and slow down page load. In the future it will not be supported.');`
-			: '');
+	let virtualModContents: string;
+	if(isClient) {
+		// Empty exports is probably enough to break the build.
+		// If for some reason its being imported as a side-effect ala `import 'astro:content';` then throw an error message.
```

Comment:
If is is imported dynamically like `await import('astro:content');` it won't fail the build either.